File: apps/web/src/api/permission.ts
import request from '@/utils/request'

export interface Permission {
  id: number
  code: string
  name: string
  type: string
  parentId: number
  path: string
  icon: string
  sortOrder: number
  children?: Permission[]
}

// 获取权限树
export function getPermissionTree() {
  return request.get<any, Permission[]>('/admin/permissions/tree')
}

// 获取所有权限
export function getAllPermissions() {
  return request.get<any, Permission[]>('/admin/permissions')
}

// 根据角色ID查询权限列表
export function getPermissionsByRoleId(roleId: number) {
  return request.get<any, Permission[]>(`/admin/permissions/role/${roleId}`)
}

File: services/admin-service/src/main/java/com/shanghaiwind/admin/entity/Permission.java
package com.shanghaiwind.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 权限实体
 */
@Data
@TableName("sys_permission")
public class Permission {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 权限编码
     */
    private String code;

    /**
     * 权限名称
     */
    private String name;

    /**
     * 类型: MENU, BUTTON, DATA
     */
    private String type;

    /**
     * 父权限ID
     */
    private Long parentId;

    /**
     * 路径
     */
    private String path;

    /**
     * 图标
     */
    private String icon;

    /**
     * 排序
     */
    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
